### PR TITLE
Bump rest-client to 2.1.0

### LINKED ISSUE #55: Use the latest version of rest-client (why a downgrade!?)
Hello,

**Why is there a downgrade of the rest-client gem?** Your commit message is empty...

Please clarify the motivation and **fix this in order to use the latest version**. It is important to use always the latest gem version to avoid bugs and security issues.

### PR DESCRIPTION
**Pull request by loomchild**
Allow rest-client dependency 2.1.0. 

Closes #55

Related to #43 and #50

### CODE REVIEW

File: lib/chargebee/rest.rb [hunk 1]
```diff
@@ -55,7 +55,12 @@ def self.request(method, url, env, params=nil, headers={})
             raise IOError.new("IO Exception when trying to connect to chargebee with url #{opts[:url]} . Reason #{e}",e)        
       end
       rheaders = response.headers
+
       rbody = response.body
+      if headers.keys.any? { |k| k.downcase == 'accept-encoding' } && rheaders[:content_encoding] == 'gzip' && rbody && !rbody.empty?
+          rbody = Zlib::GzipReader.new(StringIO.new(rbody)).read
+      end
```
Comment (loomchild) [inline lib/chargebee/rest.rb:62]: Manually decode the gzipped response if `Accept-Encoding` header was manually set and response containts `Content-Encoding: gzip`.

**Comment by loomchild** [comment]
Please advise what to do with the above warning (I don't have access to detailed message, so it's not completely clear to me what is wrong)

**Comment by loomchild** [comment]
@cb-alish Please let me know if you need anything else regarding this patch (e.g. supporting deflate, reformating code, etc.)

**Comment by cb-sriramthiagarajan** [comment]
Hi @loomchild, my apologies for the delay in response. We'll look into this and come back with any questions / required changes later this. Thanks for your patience.

**Comment by cb-sriramthiagarajan** [comment]
Hi @loomchild (and others watching this), we're still looking into this and get back soon.

While we're testing this, one of the approaches that we're thinking is to get rid of `rest-client` entirely in favour of using [Net::HTTP](https://docs.ruby-lang.org/en/3.0/Net/HTTP.html) to keep things simple. If you have any feedback on this, please do share. Thanks.

**Comment by cb-alish** [comment]
Hi @loomchild, could you take a look at #85 when you have a moment? We're transitioning to Net::HTTP with the goal of removing rest-client as a dependency. Just want to make sure everything looks good from your side. Thanks!

**Comment by loomchild** [comment]
I found an alternative solution, so I am no longer waiting for this PR. Unfortunately I won't find the time to test your new solution. Feel free to drop it if it's no longer useful.

**Pull request closed without merging**